### PR TITLE
js suffix added to main path in package.json

### PR DESCRIPTION
**Pull request by dobosalparbc**
## Description
I am getting this error using eslint
`Error: Error while loading rule 'import/no-unused-modules': Cannot find module '/Users/alpardobos/Projects/brokerchooser.com/node_modules/recharts/lib/index'. Please verify that the package.json has a valid "main" entry`

## Related Issue
https://github.com/recharts/recharts/issues/2858

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

**Comment by ckifer** [comment]
Oops, didn't even know this was a thing. Can you also add `.js` to `module` and `esnext:main` entries?

**Comment by dobosalparbc** [comment]
> Oops, didn't even know this was a thing. Can you also add `.js` to `module` and `esnext:main` entries?

Done, thank you

**Comment by ckifer** [comment]
If you want you could also make a separate PR to the `master` branch. It'll be included in the next patch in that case (otherwise not until 3.0 comes out)

**Review by ckifer — APPROVED** [review]
LGTM, thanks

**Comment by ckifer** [comment]
Ah, you'll need to update the expected output in  scripts/buildOutput.test.ts

**Comment by ckifer** [comment]
Something still going on here...

File: scripts/buildOutput.test.ts [hunk 1]
```diff
@@ -28,17 +28,17 @@ describe('expected folder structure', () => {
     });
 
     it('should point to a main file', () => {
-      expect(packageJson.main).toEqual('lib/index');
+      expect(packageJson.main).toEqual('lib/index.js');
       expect(existsSync(`./${packageJson.main}.js`)).toBe(true);
```
[the remark this answers is not in the document]
Comment (dobosalparbc) [reply scripts/buildOutput.test.ts]: Oh gosh :) Thank you!

**Review by ckifer — APPROVED** [review]

**Pull request merged**